mantidimaging/gui/windows/spectrum_viewer/view.py

```diff
@@ -371,3 +435,30 @@ def set_binning_visibility(self) -> None:
         self.bin_size_spinBox.setHidden(hide_binning)
         self.bin_step_label.setHidden(hide_binning)
         self.bin_step_spinBox.setHidden(hide_binning)
+
+    def set_roi_properties(self) -> None:
+        current_roi = self.presenter.model.get_roi(self.current_roi)
+        self.roiPropertiesGroupBox.setTitle(self.roiPropertiesGroupBoxTitle + self.current_roi)
+        roi_iter_order = ["Left", "Top", "Right", "Bottom"]
+        row = 0
+        for pos in current_roi.__iter__():
+            with QSignalBlocker(self.roiPropertiesSpinBoxes[roi_iter_order[row]]):
+                self.roiPropertiesSpinBoxes[roi_iter_order[row]].setValue(pos)
+            row = row + 1
+        self.set_roi_spinbox_ranges()
+        self.presenter.redraw_spectrum(self.current_roi)
+        self.roiPropertiesLabels["Width"].setText(str(current_roi.width))
+        self.roiPropertiesLabels["Height"].setText(str(current_roi.height))
+
+    def adjust_roi(self) -> None:
+        roi_iter_order = ["Left", "Top", "Right", "Bottom"]
+        new_points = [self.roiPropertiesSpinBoxes[prop].value() for prop in roi_iter_order]
+        new_roi = SensibleROI().from_list(new_points)
+        self.presenter.model.set_roi(self.current_roi, new_roi)
+        self.spectrum.adjust_roi(new_roi, self.current_roi)
```

Comment:
Should this be: 
```suggestion
        self.adjust_roi(new_roi, self.current_roi)
```

Reply:
No, this is in self.adjust_roi() so that would be circular. It should be `self.spectrum_widget.adjust_roi(new_roi, self.current_roi)` to adjust the spectrum roi in the spectrum widget